app/src/test/java/org/simple/clinic/home/patients/PatientsScreenControllerTest.kt

```diff
@@ -473,4 +479,23 @@ class PatientsScreenControllerTest {
         listOf(RuntimePermissionResult.DENIED, false),
         listOf(RuntimePermissionResult.NEVER_ASK_AGAIN, false))
   }
+
+  @Test
+  fun `sync indicator should be visible only when user is approved for syncing`() {
+    whenever(userSession.loggedInUser()).thenReturn(Observable.just(PatientMocker.loggedInUser(status = UserStatus.WAITING_FOR_APPROVAL).toOptional()))
+    whenever(userSession.refreshLoggedInUser()).thenReturn(Completable.never())
+    whenever(hasUserDismissedApprovedStatus.asObservable()).thenReturn(Observable.just(false))
+    whenever(hasUserDismissedApprovedStatus.get()).thenReturn(false)
+    whenever(patientSummaryResult.get()).thenReturn(PatientSummaryResult.NotSaved)
+
+    uiEvents.onNext(ScreenCreated())
+
+    canSyncStream.onNext(false)
+    canSyncStream.onNext(true)
+    canSyncStream.onNext(true)
```

Comment:
Nice! You're also verifying that multiple emissions with the same value do not have any effect.